projects/dotcms/src/lib/api/tests/DotApiForm.spec.ts

```diff
@@ -0,0 +1,52 @@
+import { DotApiForm } from '../DotApiForm';
+import { DotCMSContentTypeField } from 'dotcms/lib/models';
+
+function validateForm(formScript: string): void {
+    expect(formScript.includes('label="field1"')).toEqual(true);
+    expect(formScript.includes('hint="hint1"')).toEqual(true);
+    expect(formScript.includes('value="defaultValue1"')).toEqual(true);
+    expect(formScript.includes('required')).toEqual(true);
+}
+
+const fieldReturned = [
+    {
+        clazz: 'a.ImmutableTextField',
+        defaultValue: 'defaultValue1',
+        hint: 'hint1',
+        name: 'field1',
+        required: true,
+        value: 'value1',
+        variable: 'field1'
+    }
+];
+
+class DotApiContentTypeMock {
+    getFields(): Promise<DotCMSContentTypeField[]> {
+        return new Promise((resolve) => {
+            resolve(Array.from(fieldReturned));
+        });
+    }
+}
+
+describe('DotApiForm', () => {
+    let dotApiContentType;
+    let dotApiForm;
+
+    beforeEach(() => {
+        dotApiContentType = new DotApiContentTypeMock();
+    });
+
+    it('should render a Form', async () => {
+        const config = { identifier: '321', fields: ['field1', 'field2'] };
+        const container = document.createElement('div');
+        dotApiForm = new DotApiForm(dotApiContentType, config);
+        dotApiForm.render(container).then(() => {
+            expect(container.innerHTML.includes('<script type="module">')).toEqual(true);
```

Comment:
Also... we should not only validate the inner HTML we should validate that it used `append` to add the content to the container because is someone remove the append and does `innerHTML = ` it breaks the functionality but the tests will keep passing.